lib/webvalve/service_url_converter.rb

```diff
@@ -1,20 +1,77 @@
+require 'addressable/uri'
+require 'addressable/template'
+
 module WebValve
+  # @api private
   class ServiceUrlConverter
-    TOKEN_BOUNDARY_CHARS = Regexp.escape('.:/?#@&=').freeze
-    WILDCARD_SUBSTITUTION = ('[^' + TOKEN_BOUNDARY_CHARS + ']*').freeze
-    URL_PREFIX_BOUNDARY = ('[' + TOKEN_BOUNDARY_CHARS + ']').freeze
-    URL_SUFFIX_PATTERN = ('((' + URL_PREFIX_BOUNDARY + '|(?<=' + URL_PREFIX_BOUNDARY + ')).*)?\z').freeze
-
-    attr_reader :url
 
     def initialize(url:)
       @url = url
     end
 
-    def regexp
-      regexp_string = Regexp.escape(url)
-      substituted_regexp_string = regexp_string.gsub('\*', WILDCARD_SUBSTITUTION)
-      %r(\A#{substituted_regexp_string}#{URL_SUFFIX_PATTERN})
+    def template
+      if @url.is_a?(String)
+        protocol, working_url = split_protocol(double_stars(@url))
+        uri = Addressable::URI.parse(working_url)
```

Comment:
`Addressable::URI` will not tolerate a scheme with an asterisk in it. We don't need it to, though, because the only reason we're URI parsing this is so we can determine whether there are path or query parts to this URI so we can munge them and/or append suffixes to account for extending the URL. So we lop off protocol (an ad hoc term I'm naming `scheme + ':'` and then readd it later, because the remaining bits are still valid a `Addressable::URI`.

Reply:
okay! seems good to me.